AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -313,6 +313,42 @@ def get_type_ids(component_names: list, entity_type: EditorEntityType = EditorEn
             bus.Broadcast, "FindComponentTypeIdsByEntityType", component_names, entity_type.value)
         return type_ids
 
+    def is_property_visible(self, component_property_path: str, expected: bool = True) -> bool:
+        """
+        Used to determine if a property is visible.
+        param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :param expected: The expected visibility (default True)
+        :return: None
+        """
+        visible_key = 1
+
+        get_component_property_outcome = self.get_property_type_visibility()
+        result = get_component_property_outcome[component_property_path][visible_key]
+
+        assert (result == "Visible") == expected, \
+            f"Failure: Property visibility was not {expected} for" \
+            f"{self.get_component_name()} : {component_property_path}"
+
+        return result
```

Comment:
result is a string that could be one of 4 values including "Visible". by returning result here you are not returning the Boolean promised

Reply:
for reference sake the 4 values that visibility can return for this string from .\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\PropertyTreeEditor\PropertyTreeEditor.cpp
"NotVisible";
"Visible";
"ShowChildrenOnly";
"HideChildren";

Reply:
this is typed as  `-> bool:`
docstring `:return: None`
implemented `return result` which is string
is_property_visible implies that it answers the question (True/False) property specified is visible
`return result == "Visible"` would eliminate the need for an expected parameter
also keep in mind that `get_property_type_visibility()` is using a regex which may not capture every possible property path name so try catch key error may be required if the property path is not in the dictionary returned.